### PR TITLE
Fix missing streams

### PR DESCRIPTION
**Pull request by karthikeyann**
fix missing `stream` argument in default argument of functions.
And also in some cases, `mr` on returned objects creation.

This cleanup is done as a follow up after PR https://github.com/rapidsai/cudf/pull/9679

Almost all of libcudf functions usages of stream arg are cleaned up.
Missing `mr` still might need another clean up.

### CODE REVIEW

File: cpp/src/groupby/hash/groupby.cu [hunk 1]
```diff
@@ -308,13 +308,13 @@ class hash_compound_agg_finalizer final : public cudf::detail::aggregation_final
...
 
-    auto values_view = column_device_view::create(col);
-    auto sum_view    = column_device_view::create(sum_result);
-    auto count_view  = column_device_view::create(count_result);
+    auto values_view = column_device_view::create(col, stream);
+    auto sum_view    = column_device_view::create(sum_result, stream);
+    auto count_view  = column_device_view::create(count_result, stream);
 
     auto var_result = make_fixed_width_column(
       cudf::detail::target_type(result_type, agg.kind), col.size(), mask_state::ALL_NULL, stream);
```
Comment (nvdbaranec) [inline cpp/src/groupby/hash/groupby.cu:316]: Wondering if this `make_fixed_width_column` should take an `mr` as well.

File: cpp/src/unary/cast_ops.cu [hunk 2]
```diff
@@ -290,7 +290,8 @@ struct dispatch_unary_cast_to {
                                      rmm::cuda_stream_view stream,
                                      rmm::mr::device_memory_resource* mr)
   {
-    if (input.type() == type) return std::make_unique<column>(input);  // TODO add test for this
+    if (input.type() == type)
+      return std::make_unique<column>(input, stream, mr);  // TODO add test for this
```
Comment (nvdbaranec) [inline cpp/src/unary/cast_ops.cu:294]:
```suggestion
    if (input.type() == type){
      return std::make_unique<column>(input, stream, mr);  // TODO add test for this
    }
```

**Review by nvdbaranec — CHANGES_REQUESTED** [review]
I am terrified at how much you uncovered.

**Review by vyasr — APPROVED** [review]
Nice finds! Unfortunately I don't see a way to avoid this problem from recurring without removing default stream parameters as recommended in #9854.

File: cpp/src/groupby/hash/groupby.cu (hunk 1 above)
Comment (karthikeyann) [reply cpp/src/groupby/hash/groupby.cu:316]: `var_result` is added to `sparse_results`  which is a temporary. It's not returned. So `mr` is not needed here.

**Review by nvdbaranec — APPROVED** [review]

**Comment by karthikeyann** [comment]
@gpucibot merge

**Pull request merged**